src/lib/util/logger/Logger.ts

```diff
@@ -0,0 +1,195 @@
+import type { AutocompleteContext } from '../../structures/contexts/AutocompleteContext';
+import type { CommandContext } from '../../structures/contexts/CommandContext';
+import type { ComponentContext } from '../../structures/contexts/ComponentContext';
+import type { Command } from '../../structures/Command';
+import type { Component } from '../../structures/Component';
+import type { Listener } from '../../structures/Listener';
+import type { Plugin } from '../../structures/Plugin';
+
+export enum LoggerEvents {
+	'HANDLER_RUN' = 'handlerRun',
+	'HANDLER_ERROR' = 'handlerError',
+	'COMMAND_HANDLER_RUN' = 'commandHandlerRun',
+	'COMMAND_HANDLER_ERROR' = 'commandHandlerError',
+	'AUTOCOMPLETE_HANDLER_RUN' = 'autoCompleteHandlerRun',
+	'AUTOCOMPLETE_HANDLER_ERROR' = 'autoCompleteHandlerError',
+	'COMPONENT_HANDLER_RUN' = 'componentHandlerRun',
+	'COMPONENT_HANDLER_ERROR' = 'componentHandlerError',
+	'COMMAND_REGISTERED' = 'commandRegistered',
+	'COMMAND_UNREGISTERED' = 'commandUnregistered',
+	'COMPONENT_REGISTERED' = 'componentRegistered',
+	'COMPONENT_UNREGISTERED' = 'componentUnregistered',
+	'LISTENER_REGISTERED' = 'listenerRegistered',
+	'LISTENER_UNREGISTERED' = 'listenerUnregistered',
+	'PLUGIN_REGISTERED' = 'pluginRegistered',
+}
+
+export interface LoggerEventsInterface {
+	'handlerRun': (ctx: AutocompleteContext | CommandContext | ComponentContext) => void;
+	'handlerError': (ctx: AutocompleteContext | CommandContext | ComponentContext, error: any) => void;
+	'commandHandlerRun': (ctx: CommandContext) => void;
+	'commandHandlerError': (ctx: CommandContext, error: any) => void;
+	'autoCompleteHandlerRun': (ctx: AutocompleteContext) => void;
+	'autoCompleteHandlerError': (ctx: AutocompleteContext, error: any) => void;
+	'componentHandlerRun': (ctx: ComponentContext) => void;
+	'componentHandlerError': (ctx: ComponentContext, error: any) => void;
+	'commandRegistered': (command: Command) => void;
+	'commandUnregistered': (command: Command) => void;
+	'componentRegistered': (component: Component) => void;
+	'componentUnregistered': (component: Component) => void;
+	'listenerRegistered': (listener: Listener) => void;
+	'listenerUnregistered': (listener: Listener) => void;
+	'pluginRegistered': (plugin: Plugin) => void;
+}
+
+export const enum LogLevel {
+	TRACE = 1,
+	DEBUG = 2,
+	INFO = 3,
+	TIME = 4,
+	WARN = 5,
+	ERROR = 8,
+	OFF = 99,
+}
+
+export type LogMethods = 'trace' | 'debug' | 'info' | 'time' | 'warn' | 'timeEnd' | 'error';
+
+export declare interface ILogger {
+	on<U extends keyof LoggerEventsInterface>(
+	  event: U, listener: LoggerEventsInterface[U]
+	): this;
+  
+	emit<U extends keyof LoggerEventsInterface>(
+	  event: U, ...args: Parameters<LoggerEventsInterface[U]>
+	): boolean;
+}
+
+export class ILogger extends EventEmitter {
+	TRACE: LogLevel.TRACE;
+	DEBUG: LogLevel.DEBUG;
+	INFO: LogLevel.INFO;
+	TIME: LogLevel.TIME;
+	WARN: LogLevel.WARN;
+	ERROR: LogLevel.ERROR;
+	OFF: LogLevel.OFF;
+	level: LogLevel;
```

Comment:
```suggestion
	level: LogLevel = LogLevel.TRACE;
```

And remove the `level parameter` its useless.